### PR TITLE
OVS-4331 - More DAL speedup

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: ovs/dal/datalist.py [hunk 1]
```diff
@@ -212,30 +212,12 @@ def _execute_query(self):
...
-                        current_fields = cache_list.get(self._key, [])
-                        current_fields = list(set(current_fields + ['__all'] + invalidations[class_name]))
-                        cache_list[self._key] = current_fields
-                        self._persistent.set(key, cache_list, transaction=transaction)
-                    self._persistent.apply_transaction(transaction)
-                    successful = True
-                except AssertException:
-                    pass
+            transaction = self._persistent.begin_transaction()
+            for class_name in invalidations:
```
Comment (kvanhijf) [inline ovs/dal/datalist.py]: `for class_name, fields in invalidations.iteritems()`

File: ovs/dal/datalist.py [hunk 2]
```diff
@@ -258,21 +240,13 @@ def _execute_query(self):
...
-                            invalidated = True
-                    except KeyNotFoundException:
-                        invalidated = True
-
-                # If the key under which the list should be saved was already invalidated since the invalidations
-                # were saved, the returned list is most likely outdated. This is OK for this result, but the list
-                # won't get cached
-                if invalidated is False:
-                    self._volatile.set(self._key, self._guids, 300 + randint(0, 300))  # Cache between 5 and 10 minutes
+                    key = '{0}_{1}|{{0}}'.format(DataList.CACHELINK, class_name)
```
Comment (kvanhijf) [inline ovs/dal/datalist.py]: Shouldn't key end with `|`?

File: ovs/dal/dataobject.py [hunk 3]
```diff
@@ -666,9 +638,10 @@ def save(self, recursive=False, skip=None, _hook=None):
                 raise ObjectNotFoundException('{0} with guid \'{1}\' was deleted'.format(
                     self.__class__.__name__, self._guid
                 ))
-            except AssertException:
+            except AssertException as ex:
+                last_assert = ex
                 optimistic = False
-                pass
+                time.sleep(randint(0, 25) / 100.0)
```
Comment (kvanhijf) [inline ovs/dal/dataobject.py:643]: This doesn't make sense

Comment (khenderick) [inline ovs/dal/dataobject.py:643]: It does, it tries to reduce future assert errors.

**Pull request merged**